src/main/java/com/cloudbees/jenkins/support/filter/ContentMapping.java

```diff
@@ -49,32 +48,45 @@
     private static final String ALT_SEPARATOR = " » ";
 
     private final String original;
-    private final Pattern pattern;
     private final String replacement;
     private final int hashCode;
 
-    private ContentMapping(@Nonnull String original, @Nonnull Pattern pattern, @Nonnull String replacement) {
+    private final String[] originals;
+    private final String[] replacements;
+
+    private ContentMapping(@Nonnull String original, @Nonnull String replacement) {
```

Comment:
Switch to custom replacer, instead of RegExp, to get some more performance, so no need to use the pattern (private)